packages/astro/src/core/app/pipeline.ts

```diff
@@ -11,9 +12,28 @@ import type { SinglePageBuiltModule } from '../build/types.js';
 import { RedirectSinglePageBuiltModule } from '../redirects/component.js';
 import { createModuleScriptElement, createStylesheetElementSet } from '../render/ssr-element.js';
 import { findRouteToRewrite } from '../routing/rewrite.js';
+import { sequence } from '../middleware/index.js';
+import { createOriginCheckMiddleware } from './middlewares.js';
+import { NOOP_MIDDLEWARE_FN } from '../middleware/noop-middleware.js';
 
 export class AppPipeline extends Pipeline {
 	#manifestData: ManifestData | undefined;
+	resolvedMiddleware: MiddlewareHandler | undefined = undefined;
+
+	async getMiddleware(): Promise<MiddlewareHandler> {
```

Comment:
Is there a reason why this logic is duplicated with the container pipeline? Could it be part of `Pipeline`?